libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineTiles.kt

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.api.routetiles.v1.MapboxRouteTiles
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.geojson.BoundingBox
+import okhttp3.ResponseBody
+import retrofit2.Callback
+
+/**
+ * This is a wrapper class for the [MapboxRouteTiles] class. This class takes care of
+ * interfacing with [MapboxRouteTiles] and receives a TAR file wrapped in a ResponseBody
+ * which can be handled using a
+ * [com.mapbox.services.android.navigation.v5.utils.DownloadTask].
+ */
+class OfflineTiles private constructor(
+    private val mapboxRouteTiles: MapboxRouteTiles,
+    private val version: String
+) {
+
+    /**
+     * Call when you have constructed your OfflineTiles object with your desired parameters.
+     * [Callback] must be passed into the method to handle both the response and failure.
+     *
+     * @param callback for retrofit
+     */
+    // FIXME make internal after kotlin migration
+    protected fun fetchRouteTiles(callback: Callback<ResponseBody>) {
```

Comment:
IDE is warning
> 'protected` visibility is effectively 'private' in a final class

I believe this relates to 👇 right?
`// FIXME make internal after kotlin migration`

Reply:
Yeah, it seems that only `public` part of this class is the `Builder`. Should we split it into two and have the builder publicly available and the rest `internal`?

Reply:
yep, kolint and java classes see internal/protected `funs` differently

Reply:
> it seems that only `public` part of this class is the `Builder`. Should we split it into two and have the builder publicly available and the rest `internal`?

What do you think?

Reply:
actually Class `OfflineTiles` and `val version/ fun version()` should be public as well. Make sense?

Reply:
Nah the only "public" facing class here is `OfflineTiles.Builder` the rest are only used internally (from `RouteTileDownloader` which is package-private). That's why was wondering if we should split this two, and have the builder public and the rest internal. Does that make sense?